AC-521 fixes 'data-table' ignores

clrux opened the pull request:
# AC-521

This addresses ignored a11y checks for `data-table`.

## Sandbox

https://clrux-ac-521.sandbox.edx.org

## Reviewers

- [x] @cahrens or another TNL
- [x] @cptvitamin (more of an FYI)

clrux commented:
jenkins run bokchoy

clrux commented:
jenkins run all

clrux commented:
jenkins run python

clrux commented:
@cahrens Ready for you.

cahrens commented on lms/static/coffee/fixtures/autoenrollment.html:
```diff
@@ -10,11 +10,11 @@
       <div class="file-browse btn btn-primary">
-        <span class="browse"> Browse </span>
-        <input class="file_field" id="browseBtn" name="students_list" type="file" accept=".csv"/>
+        <label for="browseBtn-auto_enroll_csv" class="browse">${_("Browse")}</label>
```
Don't translate strings in fixture files. Somehow must comment on this must have gotten lost, or was it in a different PR?

cahrens commented on common/test/acceptance/tests/lms/test_lms_instructor_dashboard.py:
```diff
@@ -60,8 +60,8 @@ def setUp(self):
     def test_instructor_dashboard_a11y(self):
         self.instructor_dashboard_page.a11y_audit.config.set_rules({
             "ignore": [
+                'checkboxgroup',  # TODO: AC-552
```
Why is checkboxgroup added here?

clrux replied:
It was triggering a failure, but isn't part of this scope of work. I've got another PR that will address it.

cahrens replied:
How is it possible that something new started failing because of this PR?

clrux replied:
Dunno. Running the a11y tests triggered this failure. I ignored it and have another upcoming ticket that will resolve these ignores.

cahrens commented on lms/templates/ccx/enrollment.html:
```diff
@@ -79,17 +79,17 @@ <h2 class="hd hd-2">${_("Batch Enrollment")}</h2>
               <td>${member.user}</td>
               <td>${member.user.email}</td>
-              <td><a class="revoke"><span class="fa fa-times-circle" aria-hidden="true"></span> ${_("Revoke access")}</a></td>
+              <td><button type="button" class="revoke"><span class="fa fa-times-circle" aria-hidden="true"></span>${_("Revoke access")}</button></td>
```
Why type=button?

clrux replied:
It won't trigger a submit or jump to the top of the page.

cahrens replied:
OK, thanks

cahrens commented on lms/templates/instructor/instructor_dashboard_2/certificates.html:
```diff
@@ -143,9 +143,9 @@ <h3 class="hd hd-3 title">${_("Certificate Generation History")}</h3>
-                    <td class="task-date"></td>
-                    <td class="task-details"></td>
+                    <th class="task-name" scope="col">${_('Task name')}</th>
+                    <th class="task-date" scope="col">${_('Date')}</th>
```
Did you figure out how to test certificates?

clrux replied:
Yes, you can see that table on AC-524. In fact, much of this work is the same as there...

cahrens replied:
As I mentioned below, please provide a link to a course on the sandbox (with this code) where certificates are enabled so we can go through that workflow.

clrux replied:
https://clrux-ac-524.sandbox.edx.org/courses/course-v1:edX+SOH101+2016/instructor#view-certificates

clrux replied:
It was a pain in the butt to get certs running (literally took all week). The code from AC-524 is very close to this, so would [that sandbox](https://clrux-ac-524.sandbox.edx.org/courses/course-v1:edX+SOH101+2016/instructor#view-certificates) work?

clrux replied:
You're really reviewing the table that shows the number of certs.

cahrens replied:
I'm mostly concerned about the line of code referencing browseBtn-bulk-exc-students. I wanted to verify that works, as I don't see any corresponding HTML file change.

clrux replied:
I just updated that. It was incorrect (yet somehow tests were passing?) and should've matched `auto_enroll_csv`.

cahrens replied:
That's why I want to make sure you are manually testing these changes. You can't count on there being adequate automated test coverage, unfortunately.

cahrens commented on lms/static/js/certificates/views/certificate_bulk_whitelist.js:
```diff
@@ -31,7 +31,7 @@
                 el: DOM_SELECTORS.bulk_exception,
                 events: {
-                    'change #browseBtn': 'chooseFile',
+                    'change #browseBtn-bulk-exc-students': 'chooseFile',
```
I'm confused about this change because I don't see that browseBtn-bulk-exc-students appears elsewhere in this PR (I would have expected the HTML to change accordingly).

cahrens commented:
@clrux please provide a link to a course on the sandbox where certificates are enabled so we can go through that workflow.

aphanse commented:
jenkins run bokchoy

clrux commented:
jenkins run bokchoy

clrux commented:
@cahrens I've cleaned up this PR to include only `data-table` work. The other stuff was really for AC-524 and I'm not sure how it got included here. I've removed it so the files only pertain to the tables.

Functionality shouldn't be changed here, only the table structure (making it correct).

clrux commented:
jenkins run a11y

cahrens commented:
👍 with the caveat that I didn't manually test. The sandbox with certificates enabled does not seem to be running this code.

Squash of course before merging.

clrux commented:
@cahrens There's nothing to test here - only tabular data/layout to review. The changes are barely noticeable (the new `th` is bold, whereas the previous `td` wasn't). No functionality was impacted as I removed all of that craziness. Thank you ma'am!

clrux commented:
jenkins run python
